src/main/java/opendcs/dao/PropertiesSqlDao.java

```diff
@@ -182,77 +194,83 @@ public void deleteProperties(String tableName, String idColumn,
+    public int readPropertiesIntoCache(String tableName, DbObjectCache<?> cache) throws DbIoException
     {
         String q = "select * from " + tableName;
-        Integer n[] = new Integer[1];
-        n[0] = 0;
+        final AtomicInteger n = new AtomicInteger(0);
         try
         {
-            doQuery(q, rs -> {
+            doQuery(q, rs ->
+            {
                 Object ob = cache.getByKey(DbKey.createDbKey(rs, 1));
                 if (ob == null)
+                {
                     return;
+                }
                 if (!(ob instanceof HasProperties))
-                    throw new SQLException(
-                        "Cannot read properties because cached object is not HasProperties");
+                {
+                    throw new SQLException("Cannot read properties because cached object is not HasProperties");
+                }
 
                 String name = rs.getString(2);
                 String value = rs.getString(3);
                 if (value == null)
+                {
                     value = "";
+                }
 
                 HasProperties hp = (HasProperties)ob;
                 hp.setProperty(name, value);
-                n[0]++;
+                n.getAndIncrement();
             });
-            return n[0];
+            return n.get();
         }
-        catch (SQLException e)
+        catch (SQLException ex)
         {
             throw new DbIoException("Error reading properties for table "
-                + tableName + ": " + e.getMessage(),e);
+                + tableName + ": " + ex.getMessage(), ex);
         }
     }
 
     @Override
-    public int readPropertiesIntoList(String tableName, List<? extends CachableHasProperties> list,
-        String whereClause)
+    public int readPropertiesIntoList(String tableName, List<? extends CachableHasProperties> list, String whereClause)
         throws DbIoException
     {
         String q = "select * from " + tableName;
         if (whereClause != null)
+        {
             q = q + " " + whereClause;
-        ResultSet rs = doQuery(q);
-        int n = 0;
+        }
+
+        final AtomicInteger n = new AtomicInteger(0);
         try
         {
-          nextProp:
-            while(rs != null && rs.next())
+            doQuery(q, rs ->
             {
                 DbKey key = DbKey.createDbKey(rs, 1);
                 for(CachableHasProperties chp : list)
+                {
                     if (chp.getKey().equals(key))
                     {
                         String name = rs.getString(2);
                         String value = rs.getString(3);
                         if (value == null)
+                        {
                             value = "";
+                        }
 
                         chp.setProperty(name, value);
-                        n++;
-                        continue nextProp;
+                        n.getAndIncrement();
+                        return;
```

Comment:
This return 'feels' like we will only get one item?

Reply:
no, that return move to the next ResultSet so it could increment from that. Honestly that whole block confuses me in design, had a jump to label originally.